### PR TITLE
HPCC-27816 Eclwatch package map not displaying properly

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/lws.config.js [hunk 1]
```diff
@@ -67,5 +67,6 @@ let rewrite = [
 
 module.exports = {
     port: 8080,
-    rewrite: rewrite
+    rewrite: rewrite,
+    stack: ['lws-basic-auth', 'lws-request-monitor', 'lws-log', 'lws-cors', 'lws-json', 'lws-compress', 'lws-rewrite', 'lws-blacklist', 'lws-conditional-get', 'lws-mime', 'lws-range', 'lws-spa', 'lws-static', 'lws-index']
```
Comment (jeclrsg) [inline esp/src/lws.config.js:71]: @GordonSmith this isn't strictly necessary for this PR. I realized the problem wasn't related to the "upload_" flag not being set for the ValidatePackage request, but rather that the body parser middleware of lws was having an issue with the size of the request. There's an existing issue in local-web-server (130), and the workaround recommended was to exclude the lws-body-parser middleware from the stack.

File: esp/src/eclwatch/PackageMapValidateWidget.js [hunk 2]
```diff
@@ -211,7 +211,7 @@ define([
                 }
             }).then(function (response) {
                 if (lang.exists("ValidatePackageResponse", response)) {
-                    var responseText = context.validateResponseToText(response.ValidatePackageResponse);
+                    var responseText = context.validateResponseToText(response.ValidatePackageResponse.Results.Result[0]);
```
Comment (GordonSmith) [inline esp/src/eclwatch/PackageMapValidateWidget.js:214]: Is it possible that there might be no entries?

File: esp/src/src-react/components/PackageMaps.tsx [hunk 3]
```diff
@@ -133,68 +130,68 @@ export const PackageMaps: React.FunctionComponent<PackageMapsProps> = ({
...
                     setErrorMessage(Exceptions?.Exception[0].Message);
                 } else {
-                    setActiveMapValidationResult(validateResponseToText(ValidatePackageResponse));
+                    setActiveMapValidationResult(validateResponseToText(Results?.Result[0]));
                 }
             })
-            .catch(err => logger.error(err))
+            .catch(err => logger.debug(err))
             ;
     }, [activeMapTarget, activeMapXml, setActiveMapValidationResult, setErrorMessage, setShowError]);
```
Comment (GordonSmith) [inline esp/src/src-react/components/PackageMaps.tsx:148]: Lint issues

File: esp/src/src-react/components/PackageMaps.tsx [hunk 4]
```diff
@@ -269,41 +287,37 @@ export const PackageMaps: React.FunctionComponent<PackageMapsProps> = ({
...
                         } else if (Exceptions?.Exception.length > 0) {
                             setShowError(true);
                             setErrorMessage(Exceptions?.Exception[0].Message);
                         }
                     })
-                    .catch(err => logger.error(err))
+                    .catch(err => logger.debug(err))
                     ;
             }
         },
```
Comment (GordonSmith) [inline esp/src/src-react/components/PackageMaps.tsx:323]: Lint issues (a bit further down)

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/eclwatch/PackageMapValidateWidget.js (hunk 2 above)
Comment (jeclrsg) [reply esp/src/eclwatch/PackageMapValidateWidget.js]: I hadn't seen any examples where that was the case on clusters that I have access to, but I replaced that with an optional chaining operator

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**